fhevm/tee_interpreter.go

```diff
@@ -282,35 +309,59 @@ func extract1Operands(op string, environment EVMEnvironment, input []byte, runSp
 	return &cp, ct, nil
 }
 
-func extract2Operands(op string, environment EVMEnvironment, input []byte, runSpan trace.Span) (*tee.TeePlaintext, *tee.TeePlaintext, *verifiedCiphertext, *verifiedCiphertext, error) {
+func extract2Operands(op string, environment EVMEnvironment, input []byte, runSpan trace.Span) (bool, *tee.TeePlaintext, any, *verifiedCiphertext, any, error) {
```

Comment:
We lose a lot of semantic information by using `any`. What's the actual types we are interested in? `tee.Plaintext` and `*big.Int`, is that right?

Should we create an interface? A sum type?

Reply:
Okay. I'll consider that part.

Reply:
@amaury1093 Could you clarify this part?

Reply:
I'm not sure 100% what you want.

Reply:
I'm also not 100% what's the best solution,but I'm pretty sure there's a better one than `any`. If you can think of something, feel free to propose. Or else I'll take a deeper look next week.

Reply:
Okay.